include/oneapi/dpl/pstl/hetero/algorithm_impl_hetero.h

```diff
@@ -1305,37 +1305,37 @@ __pattern_inplace_merge(_ExecutionPolicy&& __exec, _Iterator __first, _Iterator
 //------------------------------------------------------------------------
 // sort
 //------------------------------------------------------------------------
-template <typename _ExecutionPolicy, typename _Iterator, typename _Compare>
+template <typename _IsSync = ::std::true_type, typename _ExecutionPolicy, typename _Iterator, typename _Compare>
 oneapi::dpl::__internal::__enable_if_hetero_execution_policy<_ExecutionPolicy, void>
 __pattern_sort(_ExecutionPolicy&& __exec, _Iterator __first, _Iterator __last, _Compare __comp,
                /*vector=*/::std::true_type, /*parallel=*/::std::true_type, /*is_move_constructible=*/::std::true_type)
 {
     if (__last - __first < 2)
         return;
 
-    __par_backend_hetero::__parallel_stable_sort(
+    auto __future_obj = __par_backend_hetero::__parallel_stable_sort(
```

Comment:
I realized the issue that if we don't need to wait (basically it means `_IsSync` is `std::false_type` then we need to return that future to the user

Reply:
If we change the return type at this level, I see compilation errors. Type mismatch vs. void function is returning something. Async algorithm can call parallel sort pattern directly.

Reply:
I removed the template parameter to switch between sync and async behavior at that level. Async sort variants will call `parallel_stable_sort` pattern directly.